dropwizard-client/src/main/java/io/dropwizard/client/HttpClientBuilder.java

```diff
@@ -378,9 +396,15 @@ protected InstrumentedHttpClientConnectionManager createConnectionManager(Regist
 
         final SSLConnectionSocketFactory sslConnectionSocketFactory;
         if (configuration.getTlsConfiguration() == null) {
-            sslConnectionSocketFactory = SSLConnectionSocketFactory.getSocketFactory();
+            if (verifier == null) {
```

Comment:
If I understand correctly, setting a `HostnameVerifier` means using a `TlsConfiguration`. What do you think about extracting `TlsConfiguration` to a local mutable variable?
```java
TlsConfiguration tlsConfiguration = configuration.getTlsConfiguration();
if (tlsConfiguration == null && verifier != null) {
    tlsConfiguration = new TlsConfiguration();
}

final SSLConnectionSocketFactory sslConnectionSocketFactory;
if (tlsConfiguration == null) {
    sslConnectionSocketFactory = SSLConnectionSocketFactory.getSocketFactory();
} else {
    sslConnectionSocketFactory = new DropwizardSSLConnectionSocketFactory(tlsConfiguration,
        verifier).getSocketFactory();
}
```